article/views/article_list_create_api_view.py

```diff
@@ -2,26 +2,49 @@
 from rest_framework.generics import ListCreateAPIView
 from rest_framework import filters
 from rest_framework_simplejwt.authentication import JWTAuthentication
+from rest_framework.response import Response
+from rest_framework.permissions import IsAuthenticatedOrReadOnly
 
 from article.serializers.article_serializer import ArticleSerializer
 from config.views import BaseView
 from article.models import Article
-from rest_framework.permissions import IsAuthenticatedOrReadOnly
+from user.models import Follow
+from article.filters import TagsFilter
 
 
 class ArticleListCreateAPIView(BaseView, ListCreateAPIView):
     serializer_class = ArticleSerializer
     queryset = Article.objects.all()
     authentication_classses = [SessionAuthentication, JWTAuthentication]
     permission_classes = [IsAuthenticatedOrReadOnly]
-    filter_backends = [filters.SearchFilter]
-    search_fields = ['title','content']
+    filter_backends = [filters.SearchFilter, TagsFilter]
+    search_fields = ["title", "content"]
 
     def get(self, request, *args, **kwargs):
         """GET: /api/articles/
         Article 목록
         """
-        return self.list(request, *args, **kwargs)
+
+        queryset = self.filter_queryset(self.get_queryset())
```

Comment:
이 코드 어떻게 동작하는건지 궁금합니다. get_queryset()과 filter_queryset() 과 관련해서 설명해주실 수 있으신가요....!

Reply:
rest_framework의 코드를 그대로 가져와봅니다 ㅋㅋ

```python
def filter_queryset(self, queryset):
        """
        Given a queryset, filter it with whichever filter backend is in use.

        You are unlikely to want to override this method, although you may need
        to call it either from a list view, or from a custom `get_object`
        method if you want to apply the configured filtering backend to the
        default queryset.
        """
        for backend in list(self.filter_backends):
            queryset = backend().filter_queryset(self.request, queryset, self)
        return queryset
```


```python
def get_queryset(self):
        """
        Get the list of items for this view.
        This must be an iterable, and may be a queryset.
        Defaults to using `self.queryset`.

        This method should always be used rather than accessing `self.queryset`
        directly, as `self.queryset` gets evaluated only once, and those results
        are cached for all subsequent requests.

        You may want to override this if you need to provide different
        querysets depending on the incoming request.

        (Eg. return a list of items that is specific to the user)
        """
        assert self.queryset is not None, (
            "'%s' should either include a `queryset` attribute, "
            "or override the `get_queryset()` method."
            % self.__class__.__name__
        )

        queryset = self.queryset
        if isinstance(queryset, QuerySet):
            # Ensure queryset is re-evaluated on each request.
            queryset = queryset.all()
        return queryset
```